File: EnumExampleC#/EnumExample/ExclusionType.cs
namespace EnumExample;

public class ExclusionType : Enum, EnumWithValue<string>
{
    public static readonly ExclusionType DISTRIBUTOR = new(nameof(DISTRIBUTOR), "Distributor");
    public static readonly ExclusionType MARKET = new(nameof(MARKET), "Market");
    public static readonly ExclusionType MOBILE = new(nameof(MOBILE), "Mobile");
    public static readonly ExclusionType COMPUTER = new(nameof(COMPUTER), "Computer");

    public string Value { get; }

    public ExclusionType(string name, string value) : base(name)
    {
        Value = value;
    }

    public string StringValue => Value;
}

File: EnumExampleJava/src/main/java/gr/devian/enumexample/ExclusionType.java
package gr.devian.enumexample;

public enum ExclusionType implements EnumWithValue<String> {
    DISTRIBUTOR("Distributor"),
    MARKET("Market"),
    MOBILE("Mobile"),
    COMPUTER("Computer");

    public static ExclusionType getByValue(final String value) {
        return EnumUtils.getEnumByValue(ExclusionType.class, value);
    }

    private final String value;

    ExclusionType(final String value) {
        this.value = value;
    }

    @Override
    public String getValue() {
        return value;
    }

    @Override
    public String getStringValue() {
        return value;
    }
}
